sw/device/tests/rv_core_ibex_nmi_irq_test.c

```diff
@@ -0,0 +1,263 @@
+      &rv_core_ibex, (dif_rv_core_ibex_nmi_state_t *)&nmi_state));
+
+  // Check that the wdog nmi is enable and is cleared.
+  CHECK(nmi_state.wdog_enabled && !nmi_state.wdog_barked,
+        "Wdog nmi wrong:\n\t"
+        "wdog_enable:%x\n\twdog_raised:%"
+        "x\n",
+        nmi_state.wdog_enabled, nmi_state.wdog_barked);
+}
+
+/**
+ * Execute the alert handler nmi interrupt test.
+ */
+static void execute_alert_handler_nmi_test(void) {
+  alert_handler_config();
+  rv_plic_testutils_irq_range_enable(&plic, plic_ctx.hart_id,
+                                     kTopEarlgreyPlicIrqIdAlertHandlerClassa,
+                                     kTopEarlgreyPlicIrqIdAlertHandlerClassa);
+  CHECK_DIF_OK(
+      dif_rv_core_ibex_enable_nmi(&rv_core_ibex, kDifRvCoreIbexNmiSourceAlert));
+
+  expected_isr_handler = alert_handler_handler;
+  nmi_fired = false;
+
+  // Trigger the alert handler to escalate.
+
+  CHECK_DIF_OK(dif_pwrmgr_alert_force(&pwrmgr, kDifPwrmgrAlertFatalFault));
+
+  IBEX_SPIN_FOR(nmi_fired, 10000);
+
+  // Check that the alert handler nmi was enable and raised a alert during the
+  // nmi irq.
+  CHECK(nmi_state.alert_enabled && nmi_state.alert_raised,
+        "Alert handler nmi wrong:\n\t"
+        "alert_enable:%x\n\talert_raised:%x\n",
+        nmi_state.alert_enabled, nmi_state.alert_raised);
+
+  CHECK_DIF_OK(dif_rv_core_ibex_get_nmi_state(
+      &rv_core_ibex, (dif_rv_core_ibex_nmi_state_t *)&nmi_state));
+
+  // Check that the alert handler nmi is enable and is cleared.
+  CHECK(nmi_state.alert_enabled && !nmi_state.alert_raised,
+        "Alert handler nmi wrong:\n\t"
+        "alert_enable:%x\n\talert_raised:%x\n",
+        nmi_state.alert_enabled, nmi_state.alert_raised);
+}
+
+void ottf_external_nmi_handler(void) {
+  nmi_fired = true;
+
+  expected_isr_handler();
+  CHECK_DIF_OK(dif_rv_core_ibex_get_nmi_state(
+      &rv_core_ibex, (dif_rv_core_ibex_nmi_state_t *)&nmi_state));
+
+  CHECK_DIF_OK(dif_rv_core_ibex_clear_nmi_state(&rv_core_ibex,
+                                                kDifRvCoreIbexNmiSourceAll));
+}
+
+/**
+ * This test do the following steps:
+ * - Trigger a watchdog bark to generate a NMI.
+ * - Check rv_core_ibex's NMI interrupt register and clear the interrupt.
+ * - Repeat the previous steps with alert handler.
+ */
+bool test_main(void) {
+  mmio_region_t addr = mmio_region_from_addr(TOP_EARLGREY_RV_PLIC_BASE_ADDR);
+  CHECK_DIF_OK(dif_rv_plic_init(addr, &plic));
+
+  addr = mmio_region_from_addr(TOP_EARLGREY_RV_CORE_IBEX_CFG_BASE_ADDR);
+  CHECK_DIF_OK(dif_rv_core_ibex_init(addr, &rv_core_ibex));
+
+  addr = mmio_region_from_addr(TOP_EARLGREY_AON_TIMER_AON_BASE_ADDR);
+  CHECK_DIF_OK(dif_aon_timer_init(addr, &aon_timer));
+
+  addr = mmio_region_from_addr(TOP_EARLGREY_ALERT_HANDLER_BASE_ADDR);
+  CHECK_DIF_OK(dif_alert_handler_init(addr, &alert_handler));
+
+  addr = mmio_region_from_addr(TOP_EARLGREY_PWRMGR_AON_BASE_ADDR);
+  CHECK_DIF_OK(dif_pwrmgr_init(addr, &pwrmgr));
```

Comment:
nit: In some test cases I have seen a pattern where all these init steps are wrapped inside a `init_peripherals()` function.  

@timothytrippel  maybe this is something that we can consider for the test conventions, but really just a nit for now.

Reply:
This improves readability a lot without comments.